Automattic/sensei
# Onboarding Wizard: Add Welcome page

Fixes #3037

### Changes proposed in this Pull Request

* Add a welcome page to the Onboarding wizard
* Add a REST API endpoint to `Sensei_Onboarding`
 * This routes to a method based on a `page` parameter in the URL, and request method.
 * Current URL: `/sensei/v1/onboarding/<page>`
* Get and set usage tracking data in the appropiate REST-callable methods.
* Move `Sensei_Admin::create_pages` so we don't have to instantiate `Sensei_Admin` in REST calls. 

### Testing instructions

* Delete Courses and My courses pages. 
* Navigate to Sensei LMS > Onboarding in wp-admin
* Observe the page
* Click *"Continue"*
* *"Yes, count me in"* checkbox in the modal that pops up should be checked if the user previously opted in to usage tracking. 
* Change the checkbox value, click *"Continue"*
* Usage tracking value should change (Can be confired in Sensei LMS > Settings, General tab, last item)
* Courses and My Courses pages should be created.

### Tasks

- [x] Add page, modal
- [x] Set up custom colors
- [x] Use REST API to get & set tracking state
- [x] Create base Sensei pages if they don't exists 
- [x] Cancel onboarding with 'Not right now'
- [ ] Configure i18n, pot generation | Moving this to a dedicated card
- [x] Advance to the next step after submit | Can be added after #3085 is merged

### Deprecated Code
- `Sensei_Admin::create_page` - Replaced by `Sensei_Onboarding_Pages::create_page`
- `Sensei_Admin::create_pages` - Replaced by `Sensei_Onboarding_Pages::create_pages`

### Screenshot

![Onboarding welcome](https://user-images.githubusercontent.com/176949/80499353-2ac1b580-896d-11ea-9ea3-e0a522125917.gif)
